File: app/src/main/java/com/victoria/bleled/util/arch/base/BaseBindingFragment.java
package com.victoria.bleled.util.arch.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

public abstract class BaseBindingFragment<T extends ViewDataBinding> extends BaseFragment {
    /************************************************************
     *  Static & Global Members
     ************************************************************/


    /************************************************************
     *  UI controls & Data members
     ************************************************************/
    public T binding;

    /************************************************************
     *  Overrides
     ************************************************************/
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, getLayoutId(), container, false);
        root = binding.getRoot();
        binding.setLifecycleOwner(this);
        return root;
    }

    @Override
    public void onResume() {
        super.onResume();

        // updateLang
        binding.invalidateAll();
    }

    /************************************************************
     *  Event Handler
     ************************************************************/


    /************************************************************
     *  Helpers
     ************************************************************/
    protected void initView() {
        binding.setLifecycleOwner(this);
    }

    /************************************************************
     *  Sub Classes
     ************************************************************/
}

File: app/src/main/java/com/victoria/bleled/base/BaseBindingFragment.kt
package com.victoria.bleled.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding

abstract class BaseBindingFragment<T : ViewDataBinding> : BaseFragment() {
    /************************************************************
     * Static & Global Members
     */
    /************************************************************
     * UI controls & Data members
     */
    lateinit var binding: T

    /************************************************************
     * Overrides
     */
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View? {
        binding = DataBindingUtil.inflate(inflater, layoutId, container, false)
        root = binding.root
        binding.lifecycleOwner = this
        return root
    }

    override fun onResume() {
        super.onResume()

        // updateLang
        binding.invalidateAll()
    }

    override fun onDestroy() {
        // binding.unbind()
        super.onDestroy()
    }

    /************************************************************
     * Event Handler
     */
    /************************************************************
     * Helpers
     */
}